### PR TITLE
switch to github actions

### PR DESCRIPTION
**Pull request by fishnix**

### CODE REVIEW

File: efs/errors.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ func ErrCode(msg string, err error) error {
 
 			return apierror.New(apierror.ErrServiceUnavailable, msg, aerr)
 		default:
-			m := msg + ": " + aerr.Message()
+			m := msg + aerr.Message()
 			return apierror.New(apierror.ErrBadRequest, m, aerr)
```
Comment (tenyo) [inline efs/errors.go:233]: I think we can get rid of `m` and just have
```
return apierror.New(apierror.ErrBadRequest, msg, aerr)
```
At least that's what worked for me

**Review by tenyo — COMMENTED** [review]
Just the one comment, otherwise looks good 👍

File: efs/errors.go (hunk 1 above)
Comment (fishnix) [reply efs/errors.go]: 👍

i also changed the tests to use `errors.As()`

File: efs/errors_test.go [hunk 2]
```diff
@@ -11,7 +11,8 @@ import (
 
 func TestErrCode(t *testing.T) {
 	apiErrorTestCases := map[string]string{
-		"": apierror.ErrBadRequest,
+		"":         apierror.ErrBadRequest,
+		"unknonwn": apierror.ErrBadRequest,
```
Comment (tenyo) [inline efs/errors_test.go:15]: 👀

File: efs/errors_test.go [hunk 3]
```diff
@@ -53,11 +54,16 @@ func TestErrCode(t *testing.T) {
...
 	for awsErr, apiErr := range apiErrorTestCases {
+		expected := apierror.New(apiErr, "test error", awserr.New(awsErr, awsErr, nil))
 		err := ErrCode("test error", awserr.New(awsErr, awsErr, nil))
-		if aerr, ok := errors.Cause(err).(apierror.Error); ok {
-			t.Logf("got apierror '%s'", aerr)
-		} else {
-			t.Errorf("expected cloudwatch error %s to be an apierror.Error %s, got %s", awsErr, apiErr, err)
+
+		var aerr apierror.Error
+		if !errors.As(err, &aerr) {
```
Comment (tenyo) [inline efs/errors_test.go:61]: This seems to accomplish the same thing, should we use it everywhere else?

File: resourcegroupstaggingapi/errors_test.go [hunk 4]
```diff
@@ -11,7 +11,8 @@ import (
 
 func TestErrCode(t *testing.T) {
 	apiErrorTestCases := map[string]string{
-		"": apierror.ErrBadRequest,
+		"":         apierror.ErrBadRequest,
+		"unknonwn": apierror.ErrBadRequest,
```
Comment (tenyo) [inline resourcegroupstaggingapi/errors_test.go:15]: 👀

**Pull request merged**